src/components/ProjectSearch/index.js

```diff
@@ -20,84 +23,256 @@ import {
   getTextSearchQuery
 } from "../../selectors";
 
+import Svg from "../shared/Svg";
+import ManagedTree from "../shared/ManagedTree";
+import SearchInput from "../shared/SearchInput";
+
+import type { List } from "immutable";
+import type { Location } from "../../types";
+import type { ActiveSearchType } from "../../reducers/types";
+import type { StatusType } from "../../reducers/project-text-search";
+
 import "./ProjectSearch.css";
 
+type Match = {
+  sourceId: string,
+  line: number,
+  column: number,
+  match: string,
+  value: string,
+  text: string
+};
+type Result = {
+  filepath: string,
+  matches: Array<Match>,
```

Comment:
it looks like matches is not optional, but we assume it is optional in some places...